docs: Clarify that identity transfer is an optimization

pchaigno opened the pull request:
This small change should clarify that the identity transfer in overlay mode is an optimization and not a requirement. We have had several users confused by this paragraph asking how security identity was obtained in direct routing mode.

pchaigno commented on Documentation/concepts/networking/routing.rst:
```diff
@@ -61,7 +61,8 @@ Identity context
   the source security identity and load balancing state to perform
-  direct-server-return (DSR).
+  direct-server-return (DSR). The security identity transfer is an optimization
+  designed to avoid one identity lookup on the remote node.
```
@brb Is the same true for `load balancing state`?

brb replied:
Nice catch! After we had removed the `bpf/bpf_lb.c`, this became no longer true. Currently, we don't transfer any DSR-related state in an encapsulation metadata. However, this might change if we switch from custom IPv{4,6} options to IP-IP encapsulation to implement DSR. For now, we can remove the mentioning of LB+DSR.

pchaigno committed:
docs: Remove mention of state transfer for DSR

7796070 removed bpf_lb.o and any LB state transfers across nodes for
DSR.

Fixes: 7796070 ("bpf: Remove bpf_lb.c")
Reported-by: Martynas Pumputis <m@lambda.lt>
Signed-off-by: Paul Chaignon <paul@cilium.io>

pchaigno committed:
docs: Clarify that identity transfer is an optimization

This small change should clarify that the identity transfer in overlay
mode is an optimization and not a requirement. We have had several users
confused by this paragraph asking how security identity was obtained in
direct routing mode.

Signed-off-by: Paul Chaignon <paul@cilium.io>
